src/coreclr/jit/unwind.cpp

```diff
@@ -312,10 +312,21 @@ void Compiler::unwindEmitFuncCFI(FuncInfoDsc* func, void* pHotCode, void* pColdC
 
     assert(endOffset <= info.compTotalHotCodeSize);
 
-    eeAllocUnwindInfo((BYTE*)pHotCode, nullptr /* pColdCode */, startOffset, endOffset, unwindCodeBytes, pUnwindBlock,
-                      (CorJitFuncKind)func->funKind);
+    const bool isFunclet = (func->funKind != FUNC_ROOT);
 
-    if (pColdCode != nullptr)
+    if (isFunclet)
+    {
+        eeAllocUnwindInfo((BYTE*)pHotCode, (BYTE*)pColdCode, startOffset, endOffset, unwindCodeBytes, pUnwindBlock,
+                          (CorJitFuncKind)func->funKind);
+    }
+    else
+    {
+        // If the function is split, EH funclets are always cold, so pass in pColdCode to indicate so.
+        eeAllocUnwindInfo((BYTE*)pHotCode, nullptr /* pColdCode */, startOffset, endOffset, unwindCodeBytes,
+                          pUnwindBlock, (CorJitFuncKind)func->funKind);
+    }
```

Comment:
I'm confused by this:
1. unwindEmitFuncCFI() should never get called, so why update it?
2. Do we have no incoming (function argument) indication if the funclet has a hot part? I guess we're currently assuming a funclet is always all cold. In that case, shouldn't we just skip the entire "hot" part here, and let the cold code below execute instead?
3. It seems confusing to me that the "else" branch comment isn't in the "if" part instead.